Added support for Eddystone-UID

sestus opened the pull request:
This pull request adds support for Eddystone-UID to PyBeacon, in terms of both advertising and scanning uid beacons.

Some notes:

- I 've changed the printed message on scanning to include the beacon type (that was only printed when verbose was provided).
- I 've kept as default behavior (e.g when no arguments provided), the advertisement of the example url.
- I would propose some refactoring on PyBeacon.py, so as to adhere to the [PEP 8 style guidelines](https://www.python.org/dev/peps/pep-0008/). I didn't want to make such changes in the context of this pull request - we could open a new Issue/pull request for this if you agree...

Please review / test my PR, and if you have any concerns or spot any issues please let me know so that I can fix them!

Thx!

PrabhanshuAttri commented:
Awesome. Will test it on the weekend and will let you know.

PrabhanshuAttri commented on PyBeacon/PyBeacon.py:
```diff
@@ -103,13 +116,38 @@ def encodeurl(url):
-def encodeMessage(url):
-    encodedurl = encodeurl(url)
-    encodedurlLength = len(encodedurl)
+def encode_uid(uid):
```
Let's use camelCase everywhere.

PrabhanshuAttri commented on PyBeacon/PyBeacon.py:
```diff
@@ -198,13 +236,22 @@ def onUrlFound(url):
 foundPackets = set()
 
+
+def on_uid_found(bytearray):
```
same camelCase

sestus commented:
I 've addressed your review comments, and also added some code to handle the user running:
`PyBeacon -u <URL> -i <UID>`  - hcitool can't advertise both at the same time.

The default behavior still remains the same, e.g running just `PyBeacon` - with no args - advertises the default url.

Thanks!

PrabhanshuAttri commented:
I was engaged with something, will look into it on this weekend. Good work

PrabhanshuAttri commented:
@sestus Could you please add few test examples for this?

PrabhanshuAttri commented:
@sestus You can work on refactoring now.
